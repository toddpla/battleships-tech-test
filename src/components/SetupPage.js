import React from 'react'

export class SetupPage extends React.Component {
  state = {
    player_one_name: '',
    player_two_name: '',

  }

  handleInputChange = (event) => {
    this.setState({[event.target.id]: event.target.value})
  }

  handleSubmit(event) {
    event.preventDefault();
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="player_one_name">Player One Name</label>
        <input id="player_one_name" type="text" onChange={this.handleInputChange} />
        <label htmlFor="player_two_name">Player Two Name</label>
        <input id="player_two_name" type="text" onChange={this.handleInputChange} />
        <input type="submit" value="Submit"/>
      </form>
    )
  }
}

export default SetupPage;
