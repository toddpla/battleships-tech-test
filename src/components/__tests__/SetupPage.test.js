import React from 'react'
import { mount } from 'enzyme'
import { SetupPage } from '../SetupPage'
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const setupScreen = renderer.create(<SetupPage />).toJSON();
  expect(setupScreen).toMatchSnapshot();
});

test('Fill names and submit should redirect to Game', () => {
  const setupScreen = mount(<SetupPage />)
  // setupScreen.find('#player_one_name').
})
