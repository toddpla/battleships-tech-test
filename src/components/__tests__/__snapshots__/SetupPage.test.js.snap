// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="player_one_name"
  >
    Player One Name
  </label>
  <input
    id="player_one_name"
    onChange={[Function]}
    type="text"
  />
  <label
    htmlFor="player_two_name"
  >
    Player Two Name
  </label>
  <input
    id="player_two_name"
    onChange={[Function]}
    type="text"
  />
  <input
    type="submit"
    value="Submit"
  />
</form>
`;
